### PR TITLE
[4.0] Clean up Factory calls in modules

### PR DESCRIPTION
**Pull request by SharkyKZ**
### Summary of Changes

Use injected application.

### Testing Instructions

Modules work like before.

### Documentation Changes Required

No.

### CODE REVIEW

**Comment by brianteeman** [comment]
Where is it injected from?

**Comment by SharkyKZ** [comment]
In `Joomla\CMS\Helper\ModuleHelper::renderModule()`. All modules have application assigned to `$app`.

**Comment by brianteeman** [comment]
Thankyou

**Comment by SharkyKZ** [comment]
Correction, this is now done in `Joomla\CMS\Dispatcher\AbstractModuleDispatcher`. Doesn't change anything here though.

**Comment by wilsonge** [comment]
Thanks!

File: administrator/modules/mod_login/Helper/LoginHelper.php [hunk 1]
```diff
@@ -47,7 +46,7 @@ function ($a, $b)
 		);
 
 		// Fix wrongly set parentheses in RTL languages
-		if (Factory::getLanguage()->isRtl())
+		if ($app->getLanguage()->isRtl())
```
Comment (Quy) [inline administrator/modules/mod_login/Helper/LoginHelper.php:49]: > Undefined variable: app in \administrator\modules\mod_login\Helper\LoginHelper.php on line 49

To reproduce:
- Install French language
- Log out

Comment (SharkyKZ) [reply administrator/modules/mod_login/Helper/LoginHelper.php:49]: Damn, how did this file get in here 😕

Comment (SharkyKZ) [reply administrator/modules/mod_login/Helper/LoginHelper.php:49]: See #25695.

**Pull request merged**